src/v/http/client.cc

```diff
@@ -467,6 +462,13 @@ ss::future<iobuf> client::response_stream::recv_some() {
           auto bufseq = iobuf_to_constbufseq(_buffer);
           boost::beast::error_code ec;
           size_t noctets = _parser.put(bufseq, ec);
+          if (!_is_header_done && _parser.is_header_done()) {
+              _is_header_done = true;
+              vlog(
+                _ctxlog.trace,
+                "response headers done: {:u}",
+                static_cast<const client::response_header&>(_parser.get()));
```

Comment:
Note: We are not redacting any headers here. I'm not sure we need to but highlighting the fact in case reviewers have an opinion on this.